### PR TITLE
brooks jeffrey scraper updates -- needed for adding many new brooks jeffrey regions

### PR DESCRIPTION
**Pull request by terinpw**
In HTML data extractor - enforce that keys aren't read if they are a key to ignore (even if a substring of the key to ignore is a key we care about)

In brooks jeffrey scraper
- add support for multiple bonds being scraped
- add support for 'No Bond Available' being scraped.

### CODE REVIEW

File: recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py [hunk 1]
```diff
@@ -43,53 +46,94 @@ def get_more_tasks(self, content, task: Task) -> List[Task]:
...
-
-    def populate_data(self, content, task: Task,
-                      ingest_info: IngestInfo) -> Optional[ScrapedData]:
+    def populate_data(self, content, task: Task, ingest_info: IngestInfo) \
+            -> Optional[ScrapedData]:
         data_extractor = HtmlDataExtractor(self.mapping_filepath)
         ingest_info = data_extractor.extract_and_populate_data(content,
                                                                ingest_info)
+        person = more_itertools.one(ingest_info.people)
+        booking = person.bookings[0]
```
Comment (jamwalla) [inline recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py]: should this be `one(person.bookings)` too?

File: recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py [hunk 2]
```diff
@@ -43,53 +46,94 @@ def get_more_tasks(self, content, task: Task) -> List[Task]:
...
+        if split_bonds or bond_status:
+            bond = charge.create_bond()
+            if split_bonds:
+                bond.amount = split_bonds[idx]
+            if bond_status:
+                bond.status = bond_status.value
+
+
+def _split(repeated_object_str: str) -> List[str]:
+    return list(filter(None, repeated_object_str.split('\n')))
```
Comment (jamwalla) [inline recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py]: nit: `[s for s in repeated_object_str.split('\n') if s]` might be more readable

File: recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py [hunk 3]
```diff
@@ -43,53 +46,94 @@ def get_more_tasks(self, content, task: Task) -> List[Task]:
...
+
+        split_bonds = _split(booking.total_bond_amount)
+        if len(split_bonds) > 1:
+            return split_bonds, None
+    return None, None
+
+
+def _split_charges(
+        booking: _Booking, split_bonds: Optional[List[str]],
+        bond_status: Optional[BondStatus]) -> None:
```
Comment (jamwalla) [inline recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py]: talked offline - lets make IngestObject subclass constructors public and have this return a list of Charges

File: recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py (hunk 2 above)
Comment (terinpw) [reply recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py]: Good point, like this more.

File: recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py (hunk 3 above)
Comment (terinpw) [reply recidiviz/ingest/vendors/brooks_jeffrey/brooks_jeffrey_scraper.py]: Done!

**Pull request merged**